### PR TITLE
refactor `blas/ext/base/dcusumkbn` to follow current project conventions

### LINKED ISSUE #1472: [RFC]: refactor `blas/ext/base/dcusumkbn` to follow current project conventions
### Description

This RFC proposes refactoring [`@stdlib/blas/ext/base/dcusumkbn`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/dcusumkbn) to follow current project conventions.

These conventions are outlined in more detail in https://github.com/stdlib-js/stdlib/issues/788. Namely, we want to migrate from C++ add-on interfaces to C add-on interfaces and then make various style and simplification changes.

Some example commits which perform most, if not all, of the requested changes include https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1 (saxpy),  https://github.com/stdlib-js/stdlib/commit/3c82fe7b6c0f9fd6ef02c86d6e2024d708ac8c14 (cswap), and https://github.com/stdlib-js/stdlib/commit/5b15c9675be434c0e11b1ec1e1f78d7b089abbcd (dasum).

### Related Issues

Related issues #788, #1152.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by itskdhere**
Resolves #1472 .

## Description

> What is the purpose of this pull request?

This pull request:

-   refactors the existing implementation of `blas/ext/base/dcusumkbn` to follow the current project convention.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   fixes #1472

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/blas/ext/base/dcusumkbn/lib/ndarray.native.js [hunk 1]
```diff
@@ -41,26 +42,20 @@ var addon = require( './dcusumkbn.native.js' );
...
-	}
-	if ( strideY < 0 ) {
-		offsetY += (N-1) * strideY;
-	}
-	viewX = new Float64Array( x.buffer, x.byteOffset+(x.BYTES_PER_ELEMENT*offsetX), x.length-offsetX ); // eslint-disable-line max-len
-	viewY = new Float64Array( y.buffer, y.byteOffset+(y.BYTES_PER_ELEMENT*offsetY), y.length-offsetY ); // eslint-disable-line max-len
+	offsetX = minViewBufferIndex( N, strideX, offsetX );
+	viewX = offsetView( x, offsetX );
+	offsetY = minViewBufferIndex( N, strideY, offsetY );
+	viewY = offsetView( y, offsetY );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dcusumkbn/lib/ndarray.native.js]:
```suggestion
	offsetX = minViewBufferIndex( N, strideX, offsetX );
	offsetY = minViewBufferIndex( N, strideY, offsetY );

	viewX = offsetView( x, offsetX );
	viewY = offsetView( y, offsetY );
```

File: lib/node_modules/@stdlib/blas/ext/base/dcusumkbn/manifest.json [hunk 2]
```diff
@@ -1,40 +1,77 @@
...
+        "./include"
+      ],
+      "libraries": [
+        "-lm"
+      ],
+      "libpath": [],
+      "dependencies": []
+    }
+  ]
+}
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dcusumkbn/manifest.json]:
```suggestion
}

```

Comment (Pranavchiku) [reply lib/node_modules/@stdlib/blas/ext/base/dcusumkbn/manifest.json]: add newline at end of file

File: lib/node_modules/@stdlib/blas/ext/base/dcusumkbn/src/addon.c [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	STDLIB_NAPI_ARGV_INT64( env, sum, argv, 1 );
+	STDLIB_NAPI_ARGV_INT64( env, strideX, argv, 3 );
+	STDLIB_NAPI_ARGV_INT64( env, strideY, argv, 5 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT64ARRAY( env, X, N, strideX, argv, 2 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT64ARRAY( env, Y, N, strideY, argv, 4 );
+
+	napi_value v;
+	napi_status status = napi_create_double( env, stdlib_strided_dcusumkbn( N, sum, X, strideX, Y, strideY ), &v );
+	assert( status == napi_ok );
+	return v;
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dcusumkbn/src/addon.c]: you need not to return here `v`, it will be `nullptr`, refer `addon.cpp`

**Review by Pranavchiku — CHANGES_REQUESTED** [review]
minor changes required, once addressed this will be good to merge.

File: lib/node_modules/@stdlib/blas/ext/base/dcusumkbn/src/addon.c (hunk 3 above)
Comment (Pranavchiku) [reply lib/node_modules/@stdlib/blas/ext/base/dcusumkbn/src/addon.c]: I think this is fine, not exactly sure though.

cc @Planeshifter @kgryte

Comment (itskdhere) [reply lib/node_modules/@stdlib/blas/ext/base/dcusumkbn/src/addon.c]: fixed. 
yes, it'll return `NULL`

**Comment by itskdhere** [comment]
@Pranavchiku I've applied the suggested changes. And all the checks are now passing.

File: lib/node_modules/@stdlib/blas/ext/base/dcusumkbn/src/addon.c [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+*
+* @private
+* @param env    environment under which the function is invoked
+* @param info   callback data
+* @return       Node-API value
+*/
+static napi_value addon( napi_env env, napi_callback_info info ) {
+	STDLIB_NAPI_ARGV( env, info, argv, argc, 6 );
+	STDLIB_NAPI_ARGV_INT64( env, N, argv, 0 );
+	STDLIB_NAPI_ARGV_INT64( env, sum, argv, 1 );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dcusumkbn/src/addon.c]: This is incorrect. `sum` is a double, not an `int64`.

**Review by kgryte — CHANGES_REQUESTED** [review]
The current `addon.c` implementation is incorrect.

**Comment by itskdhere** [comment]
> The current `addon.c` implementation is incorrect.

@kgryte I've changed it from `int64` to `double`

But now the CI is failing because it looks like it can't find `@stdlib/napi/argv_double.h`
Should I revert 5d12725 ?

File: lib/node_modules/@stdlib/blas/ext/base/dcusumkbn/manifest.json [hunk 5]
```diff
@@ -1,40 +1,76 @@
...
+      "include": [
+        "./include"
+      ],
+      "libraries": [
+        "-lm"
+      ],
+      "libpath": [],
+      "dependencies": [
+        "@stdlib/napi/argv",
+        "@stdlib/napi/argv-int64",
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dcusumkbn/manifest.json:42]:
```suggestion
        "@stdlib/napi/argv-int64",
        "@stdlib/napi/argv-double",
```

**Comment by kgryte** [comment]
No. You should not revert it. You did not update the `manifest.json` accordingly to reflect those changes in `addon.c`.

File: lib/node_modules/@stdlib/blas/ext/base/dcusumkbn/manifest.json [hunk 6]
```diff
@@ -40,7 +40,6 @@
       "dependencies": [
         "@stdlib/napi/argv",
         "@stdlib/napi/argv-int64",
-        "@stdlib/napi/argv_double",
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dcusumkbn/manifest.json]: This was the wrong package name. Notice the kebabcase convention of the other packages in this list.

Comment (itskdhere) [reply lib/node_modules/@stdlib/blas/ext/base/dcusumkbn/manifest.json]: Ok, got it. Thanks for the clarification.

**Comment by itskdhere** [comment]
> No. You should not revert it. You did not update the `manifest.json` accordingly to reflect those changes in `addon.c`.

Ok. Now it's clear to me.

**Review by Planeshifter — APPROVED** [review]
LGTM; thank you!

**Pull request merged**